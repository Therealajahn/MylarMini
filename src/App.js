import MiniModular from './View/MiniModular/MiniModular.js';
import './App.css';

function App() {
    let appStyle = {
        width: '100vw',
        height: '100vh',
        display: 'grid',
        backgroundColor: '#707070'
    };
    return (
        <div className='App' style={appStyle}>
            <MiniModular/> 
        </div>
    );
};

export default App;
