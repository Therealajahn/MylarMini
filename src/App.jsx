import MiniModular from './2-View/MiniModular/MiniModular.jsx';
import Sound from './Sound/Sound.jsx';
import { useRef,useState,useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { addEventToApp } from './1-Model/controlEventSlice.jsx';
import { updateMouseLocation } from './1-Model/getEventData.jsx';



export default function App() {

		const appRef = useRef();
		const getEventArray = useSelector(state => state.controlEvent.eventType);
	  const dispatch = useDispatch();
		

   	const whenEventAdded = useEffect(() => {
			console.log('(from app)event array added: ',getEventArray);
			getEventArray.forEach(getEvent => {
				appRef.current.addEventListener(getEvent.payload,event => {
					switch(event.type){
						case 'mousemove':
							whenMouseMoves();
						break;
					}
					function whenMouseMoves() {
						dispatch(updateMouseLocation({x: event.clientX,y: event.clientY}));
						//console.log('whenMouseMoves(in whenEventAdded)',event.clientY);
					};
				});
			});
		},[getEventArray]);
		
	


    let appStyle = {
        width: '100vw',
        height: '100vh',
        display: 'grid',
        backgroundColor: '#707070'
    };
		

    return (
        <div 
					data-testid="app-div" 
					className='App' 
					style={appStyle} 
					ref={appRef}>
						<MiniModular appRef={appRef}/>
            <Sound/>
        </div>
    );
};

