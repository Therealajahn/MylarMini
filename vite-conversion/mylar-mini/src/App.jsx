import MiniModular from './View/MiniModular/MiniModular.jsx';
import Sound from './Sound/Sound.jsx';

export default function App() {
    let appStyle = {
        width: '100vw',
        height: '100vh',
        display: 'grid',
        backgroundColor: '#707070'
    };
    return (
        <div data-testid="app-div" className='App' style={appStyle}>
            <MiniModular/> 
            <Sound/>
        </div>
    );
};

