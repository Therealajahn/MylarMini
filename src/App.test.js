import { render, screen, cleanup } from '@testing-library/react';
import App from './App';

afterEach(() => {
    cleanup();
})

describe('App renders',() => {
    const renderApp = () => {
        render(
            <App/>
        )
    }  

    it("renders the app",() => {
        expect(renderApp()).toMatchSnapshot();
    })
})
