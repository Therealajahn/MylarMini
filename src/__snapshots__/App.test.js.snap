// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders renders the app 1`] = `undefined`;
